tests/Integrations/OpenAI/OpenAITest.php

```diff
@@ -0,0 +1,521 @@
+        ini_set("log_errors", 1);
+        ini_set("error_log", __DIR__ . "/openai.log");
+        self::putEnvAndReloadConfig([
+            'DD_OPENAI_LOG_PROMPT_COMPLETION_SAMPLE_RATE=1.0',
+            'DD_OPENAI_LOGS_ENABLED=true',
+            'DD_LOGS_INJECTION=true',
+            'DD_TRACE_DEBUG=true',
+            'DD_TRACE_GENERATE_ROOT_SPAN=0',
+            'DD_SERVICE=openai-test',
+            'DD_ENV=test',
+            'DD_VERSION=1.0',
+        ]);
+        if (file_exists(__DIR__ . "/openai.log")) {
+            $this->errorLogSize = (int)filesize(__DIR__ . "/openai.log");
+        } else {
+            $this->errorLogSize = 0;
+        }
+    }
+
+    protected function envsToCleanUpAtTearDown()
+    {
+        return [
+            'DD_OPENAI_SERVICE',
+            'DD_OPENAI_METRICS_ENABLED',
+            'DD_OPENAI_LOGS_ENABLED',
+            'DD_OPENAI_SPAN_CHAR_LIMIT',
+            'DD_OPENAI_SPAN_PROMPT_COMPLETION_SAMPLE_RATE',
+            'DD_OPENAI_LOG_PROMPT_COMPLETION_SAMPLE_RATE',
+        ];
+    }
+
+    protected function ddTearDown()
+    {
+        parent::ddTearDown();
+        //shell_exec("rm -f " . __DIR__ . "/openai.log");
+        $error = $this->checkErrors();
+        if ($error) {
+            $this->fail("Got error:\n$error");
+        }
+    }
+
+    private function call($resource, $openAIFn, $metaHeaders, $responseBodyArray, $openAIParameters = null, $responseCode = 200)
+    {
+        $this->isolateTracerSnapshot(fn: function () use ($resource, $openAIFn, $metaHeaders, $responseBodyArray, $openAIParameters, $responseCode) {
+            $response = new Response($responseCode, ['Content-Type' => 'application/json; charset=utf-8', ...$metaHeaders], json_encode($responseBodyArray));
+            $client = mockClient($response);
+            try {
+                if ($openAIParameters) {
+                    $client->{$resource}()->{$openAIFn}($openAIParameters);
+                } else {
+                    $client->{$resource}()->{$openAIFn}();
+                }
+            } catch (\OpenAI\Exceptions\ErrorException $e) {
+                // Ignore exceptions, they're "expected"
+            }
+        }, snapshotMetrics: true, logsFile: __DIR__ . "/openai.log");
+    }
+
+    private function callStreamed($resource, $openAIFn, $metaHeaders, $responseBodyArray, $openAIParameters = null)
+    {
+        $this->isolateTracerSnapshot(fn: function () use ($resource, $openAIFn, $metaHeaders, $responseBodyArray, $openAIParameters) {
+            $response = new Response(
+                headers: $metaHeaders,
+                body: new Stream($responseBodyArray)
+            );
+            $client = mockClient($response);
+            if ($openAIParameters) {
+                $client->{$resource}()->{$openAIFn}($openAIParameters);
+            } else {
+                $client->{$resource}()->{$openAIFn}();
+            }
```

Comment:
Please verify the response can also be actually read back by the caller - the integration must not _consume_ it.